src/Tooltip.js

```diff
@@ -79,10 +89,24 @@ class Tooltip extends React.Component {
     };
   }
 
+  show() {
+    if (!this.props.isOpen) {
+      this.toggle();
+    }
+  }
+  hide() {
+    if (this.props.isOpen) {
+      this.toggle();
+    }
+  }
+
   handleDocumentClick(e) {
     if (e.target === this._target || this._target.contains(e.target)) {
-      if (this._hoverTimeout) {
-        clearTimeout(this._hoverTimeout);
+      if (this._hideTimeout) {
+        clearTimeout(this._hideTimeout);
+      }
+      if (this._showTimeout) {
+        clearTimeout(this._showTimeout);
```

Comment:
Yeah that's true, it would run clearTimeout unnecessarily. That must have already been an issue before this PR though, because there was no unsetting of the timeout property either. But I may as well include that change in this PR just to tidy it up.